tests/ws_scaffold.py

```diff
@@ -47,18 +47,15 @@ def test(network, args):
 
 
 def run(args):
-    hosts = ["localhost"] * (4 if args.consensus == "bft" else 2)
-
     with infra.network.network(
-        hosts, args.binary_dir, args.debug_nodes, args.perf_nodes, pdb=args.pdb
+        args.nodes, args.binary_dir, args.debug_nodes, args.perf_nodes, pdb=args.pdb
     ) as network:
         network.start_and_join(args)
         test(network, args)
 
 
 if __name__ == "__main__":
-
     args = infra.e2e_args.cli_args()
     args.package = args.app_script or "liblogging"
-
+    args.nodes = infra.e2e_args.min_nodes(args, f=1)
```

Comment:
Not equivalent to what we had before, where we rather bizarrely had f=1 in bft and f=0 in cft. I think this is what we want in this case though, because we're checking transactions are committed.